drivers/at86rf215/at86rf215_internal.c

```diff
@@ -0,0 +1,210 @@
+    spi_release(SPIDEV);
+}
+
+void at86rf215_reg_write_bytes(const at86rf215_t *dev, uint16_t reg, const void *data, size_t len)
+{
+    reg = htons(reg | FLAG_WRITE);
+
+    getbus(dev);
+    spi_transfer_bytes(SPIDEV, CSPIN, true, &reg, NULL, sizeof(reg));
+    spi_transfer_bytes(SPIDEV, CSPIN, false, data, NULL, len);
+    spi_release(SPIDEV);
+}
+
+uint8_t at86rf215_reg_read(const at86rf215_t *dev, uint16_t reg)
+{
+    uint8_t val;
+
+    reg = htons(reg | FLAG_READ);
+
+    getbus(dev);
+    spi_transfer_bytes(SPIDEV, CSPIN, true, &reg, NULL, sizeof(reg));
+    val = spi_transfer_byte(SPIDEV, CSPIN, false, 0);
+    spi_release(SPIDEV);
+
+    return val;
+}
+
+void at86rf215_reg_read_bytes(const at86rf215_t *dev, uint16_t reg, void *data, size_t len)
+{
+    reg = htons(reg | FLAG_READ);
+
+    getbus(dev);
+    spi_transfer_bytes(SPIDEV, CSPIN, true, &reg, NULL, sizeof(reg));
+    spi_transfer_bytes(SPIDEV, CSPIN, false, NULL, data, len);
+    spi_release(SPIDEV);
+}
+
+#endif /* TEST_SUITES */
+
+void at86rf215_filter_ack(at86rf215_t *dev, bool on)
+{
+    /* only listen for ACK frames */
+    uint8_t val = on ? (1 << IEEE802154_FCF_TYPE_ACK)
+                     : (1 << IEEE802154_FCF_TYPE_BEACON)
+                     | (1 << IEEE802154_FCF_TYPE_DATA)
+                     | (1 << IEEE802154_FCF_TYPE_MACCMD);
+
+    at86rf215_reg_write(dev, dev->BBC->RG_AFFTM, val);
+}
+
+void at86rf215_get_random(at86rf215_t *dev, uint8_t *data, size_t len)
+{
+    at86rf215_disable_baseband(dev);
+
+    uint8_t rxbwc = at86rf215_reg_read(dev, dev->RF->RG_RXBWC);
+
+    /* The analog frontend of the radio must be configured to the
+       widest filter bandwidth; The bit RXBWC.IFS must be set to 1 */
+    at86rf215_reg_write(dev, dev->RF->RG_RXBWC, 0x1B);
+
+    while (len--) {
+        *data++ = at86rf215_reg_read(dev, dev->RF->RG_RNDV);
+    }
+
+    at86rf215_reg_write(dev, dev->RF->RG_RXBWC, rxbwc);
+    at86rf215_enable_baseband(dev);
+}
+
+uint16_t at86rf215_chan_valid(at86rf215_t *dev, uint16_t chan)
+{
+    if (is_subGHz(dev)) {
+        if (chan >= dev->num_chans) {
+            return dev->num_chans - 1;
+        }
+    } else {
+        if (chan < IEEE802154_CHANNEL_MIN) {
+            return IEEE802154_CHANNEL_MIN;
+        } else if (chan >= IEEE802154_CHANNEL_MIN + dev->num_chans) {
+            return IEEE802154_CHANNEL_MIN + dev->num_chans - 1;
+        }
+    }
+
+    return chan;
+}
+
+#include "net/gnrc/netif/internal.h"
+bool at86rf215_switch_mode(at86rf215_t *dev, uint8_t new_mode)
+{
+    if (new_mode == dev->mode) {
+        return false;
+    }
+
+    bool pdu_changed = (new_mode == AT86RF215_MODE_LEGACY_OQPSK ||
+                        dev->mode == AT86RF215_MODE_LEGACY_OQPSK);
+
+    dev->mode = new_mode;
+
+#ifdef MODULE_GNRC_IPV6 /* XXX move this out of the driver */
+    if (pdu_changed) {
+        gnrc_netif_ipv6_init_mtu(gnrc_netif_get_by_pid(thread_getpid()));
+    }
+#endif
```

Comment:
I'm sure @miri64 is not happy about this.
But catching the events that could lead to this in `gnrc_netif_set_from_netdev()` seems even worse as this would mean pushing driver-specific behavior into common code.

Is it possible for the driver to send an event for the upper layer in a stack agnostic way? 
Extending [`_event_cb`](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/netif/gnrc_netif.c#L1466) would be the first thing that's coming to my mind.